[carineang] iP

carineang opened the pull request:
# TaskBuddy

> “Your mind is for having ideas, not holding them.” – David Allen

TaskBuddy relieves your stress of managing  tasks and keeps you focused on the important things that matters.
- Text based
- ~~Hard~~ Easy to use
- _Super fast_ to learn :smiley:

## Features

- Managing tasks
- Searching for tasks using keywords
- Mark/unmark tasks
- Viewing all tasks in a list

## Installation

1. Download the app [here](https://github.com/carineang/ip)
2. Run the application using:

```bash
java -jar taskbuddy.jar
```

## Optional
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        new TaskBuddy("ip/data/taskbuddy.txt").run();
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

carineang committed:
Add chatbot rename, greet, exit messages

carineang committed:
Add echo

carineang committed:
Add store and list user input

carineang committed:
Add mark as done

carineang committed:
Add todos, events, deadlines

carineang committed:
Add A-TextUiTesting

carineang committed:
Add handle errors

carineang committed:
Add delete

carineang committed:
Refactor code

carineang committed:
Update user guide

carineang committed:
Add save

carineang committed:
Merge branch-Level-7

carineang committed:
Add date and time

carineang committed:
Merge branch-Level-8

carineang committed:
Add javadoc

carineang committed:
Merge branch-A-JavaDoc

carineang committed:
Update Level 7

carineang committed:
Update Level 8

carineang committed:
Add A-MoreOOP

carineang committed:
Merge branch-A-MoreOOP

carineang committed:
Add A-Packages

carineang committed:
Merge branch-A-Packages

carineang committed:
Merge branch 'add-gradle-support'
Add and merge add-gradle-support

carineang committed:
Add JUnit

carineang committed:
Merge branch-A-JUnit

AbiHalim commented on src/main/java/taskbuddy/Ui.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.Scanner;
+
+public class Ui {
+    private final String indent = "   ";
```
If the variable is final, should it be in SCREAMING_SNAKE_CASE?

AbiHalim commented on src/main/java/taskbuddy/command/NoOpCommand.java:
```diff
@@ -0,0 +1,16 @@
+import taskbuddy.Storage;
+import taskbuddy.Ui;
+
+public class NoOpCommand extends Command {
```
Class name could be clearer

AbiHalim commented on src/main/java/taskbuddy/TaskBuddy.java:
```diff
@@ -0,0 +1,32 @@
+package taskbuddy;
+import taskbuddy.command.*;
```
Should the imported classes be listed explicitly?

AbiHalim commented on src/main/java/taskbuddy/TaskBuddy.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    public void run() {
+        ui.printWelcomeMessage();
```
I like how you put printWelcomeMessage() into a separate method 👍

AbiHalim reviewed (commented):
LGTM. Just a few nits to fix.

carineang committed:
Add find

carineang committed:
Merge branch-Level-9

carineang committed:
Add A-Javadocs

carineang committed:
Add A-CodingStandard

carineang committed:
Merge branch-A-CodingStandard

carineang committed:
Add JavaFX GUI

carineang committed:
Update bye command

Modify ExitCommand to close the GUI.

carineang committed:
Merge branch-Level-10

carineang committed:
Update JavaDocs

Edit Javadocs for most classes.

carineang committed:
Add Assertions

carineang committed:
Merge branch-A-Assertions

Update assertions in Storage.java.

carineang committed:
Update assertions

carineang committed:
Merge pull request #1 from carineang/branch-A-Assertions

Update assertions

carineang committed:
Update Javadocs

Editted Javadocs for Launcher.java.

carineang committed:
Merge branch 'master' of https://github.com/carineang/ip

carineang committed:
Refactor Code Quality

Apply SLAP to Parser.java and Storage.java

carineang committed:
Merge pull request #3 from carineang/branch-A-CodeQuality

Refactor Code Quality

carineang committed:
Add Gradle workflow file

carineang committed:
Merge branch 'master' of https://github.com/carineang/ip

carineang committed:
Update JUnit tests

carineang committed:
Add view schedules

View tasks based on specified dates.

carineang committed:
Refactor code

Apply SLAP to Task, Todo, Deadline, Event, ViewCommand classes.

carineang committed:
Add Checkstyle

Add checkstyle.xml and suppressions.xml. Update code using checkstyle.

carineang committed:
Merge branch-A-CheckStyle

carineang committed:
Merge branch 'branch-A-CheckStyle'

carineang committed:
Update JavaFX GUI

Edit dialog box. Change window size.

carineang committed:
Add title in GUI

Set title of GUI window.

carineang committed:
Update README.md

carineang committed:
Update user guide

carineang committed:
Update user guide

Update tag location.

carineang committed:
Add Ui screenshot

Updated user guide to include the screenshot.

carineang committed:
Add streams

Added streams in TaskList.

carineang committed:
Update mark and unmark

Edit isDone variable in Storage class.

carineang committed:
Refactor code

Reduces code length for Ui class.

carineang committed:
Update exit command

Added error message for unsuccessful file saving.

carineang committed:
Update JUnit tests

Update JUnit tests for Parser and Ui classes. Add JUnit tests for TaskList class.

carineang committed:
Improve code quality

Removes deeply nested code in Storage class.

carineang committed:
Update error message

carineang committed:
Update Storage

Update load and save methods.

carineang committed:
Update Javadocs

Updated javadocs for most files.

carineang committed:
Update README.md
